R/impact_method2.R

```diff
@@ -0,0 +1,442 @@
+  tab <- merge_in(tab, target_cohortS, c(target_cohortS = "population"))
+  
+  ## 5. calculate age level fvps
+  tab$fvps <- NA
+  i <- is.na(tab$target)
+  # campaign
+  tab$fvps[!i] <- tab$coverage[!i] * tab$target[!i] / tab$target_cohortS[!i] * tab$population[!i]
+  # routine
+  tab$fvps[i] <- tab$coverage[i] * tab$population[i]
+  # report problems - fvps >> pop
+  tab$diff <- (tab$target - tab$target_cohortS) / tab$target_cohortS
+  write.csv(tab[tab$diff > 1., ], "problematic_campaign.csv", row.names = FALSE)
+  
+  ## 6. calculate age level coverage
+  tab$coverage <- tab$fvps / (tab$population+1) # plus one to avoid 0 pop for old age
+  tab <- tab[c("vaccine", "activity_type", "country", "year", "age", "gavi_support", "population", "coverage", "fvps")]
+  
+  ## cap fvps by population, and coverage by 1 - eg. AFG 2015 Measles campagin ; HTI Measles 2025
+  ## This operation makes sense - you cannot vaccinate more people than population
+  ## In addition, we need to cap coverage by 1 for method3 implementation (not happening for now)
+  i <- tab$coverage > 1. & !is.na(tab$coverage)
+  tab$coverage[i] <- 1
+  tab$fvps[i] <- tab$population[i]
+  
+  ## 8. make a local temporary table
+  ## so that we only manipulate pop-coverage-fvps once
+  DBI::dbWriteTable(con, "temporary_coverage_fvps", tab, temporary = TRUE, overwrite = TRUE)
+  message("Temporary table created - Success!")
+  
+}
+
+impact_output <- function(impact, method) {
+  ## bind output
+  impact_full <- NULL
+  impact_simplified <- NULL
+  for(i in seq_along(impact)){
+    impact_full <- rbind(impact[[i]]$impact_full, impact_full)
+    impact_simplified <- rbind(impact[[i]]$impact_simplified, impact_simplified)
+  }
+  ## order output
+  if(method == "method1") {
+    impact_full <- impact_full[order(impact_full$index, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$country, impact_simplified$year), ]
+    
+  } else {
+    impact_full <- impact_full[order(impact_full$index, impact_full$support_type, impact_full$country, impact_full$year, impact_full$age), ]
+    impact_simplified <- impact_simplified[order(impact_simplified$index, impact_simplified$impact_type, impact_simplified$support_type, impact_simplified$country, impact_simplified$year), ]
+    
+  }
+  row.names(impact_full) <- NULL
+  row.names(impact_simplified) <- NULL
+  return(list(impact_full = impact_full, impact_simplified = impact_simplified))
+}
+
+line_up <- function(recipe, group, burden_outcomes) {
+  # The line_up function will subject to change, once impact recipes are imported into montagu
+  stopifnot(nrow(recipe) == 1L)
+  ### burden_outcome
+  burden_outcome <- unlist(strsplit(recipe$burden_outcome, ","))
+  j <- duplicated(burden_outcome)
+  if(any(j)) stop("Duplicated burden_outcomes not expected.")
```

Comment:
Please reformat (and elsewhere) as

```
if (any(j)) {
  stop("Duplicated burden_outcomes not expected.")
}
```

(space after `if`, clause on newline, curly braces)

Reply:
Done.